From: Salvador Ortiz Garcia <sog@msg.com.mx>
To: dev@perl.apache.org
Date: Sat, 29 Nov 2008 04:20:51 -0600
Message-Id: <1227954051.4548.162.camel@monica.msg.com.mx>

Hi all,

This afternoon I try to compile my standard bundle (mod_perl 1.30,
apache 1.3.41, libapreq-1.33, mod_ssl-2.8.31) in Fedora 10 that ships
with gcc 4.3.2 only to discover that the change announced in 
<http://www.gnu.org/software/gcc/gcc-4.2/changes.html> broke the
compilation of apache-1.3.41/src/os/unix/os-inline.c when using -std=c99
(default for Fedora's perl)

I known that the problem is apache related, not mod_perl's, but there
are many mod_perl 1.x users out there that can be in trouble.

The attached patch fixes the problem.

Any one have contacts with legacy apache maintainers?

Regards.

Salvador Ortiz.

========= CUT ======
--- src/os/unix/os.h.orig       2008-11-29 00:11:23.000000000 -0600
+++ src/os/unix/os.h    2008-11-29 00:15:15.000000000 -0600
@@ -33,7 +33,12 @@
 /* Compiler supports inline, so include the inlineable functions as
  * part of the header
  */
+#ifdef __GNUC_STDC_INLINE__
+#define INLINE extern ap_inline __attribute__((gnu_inline))
+#else
 #define INLINE extern ap_inline
+#endif
+
 
 INLINE int ap_os_is_path_absolute(const char *file);



